### PR TITLE
8305757: Call Method::compute_has_loops_flag() when creating CDS archive

### PR DESCRIPTION
**Pull request by iklam**
A simple optimization -- call Method::compute_has_loops_flag() for all methods during CDS archive dump. This way, the JIT doesn't need to do this at runtime.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305757](https://bugs.openjdk.org/browse/JDK-8305757): Call Method::compute_has_loops_flag() when creating CDS archive

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [9f5ef50d](https://git.openjdk.org/jdk/pull/13390/files/9f5ef50d52a9658301fcea3b103137a92696c951)
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13390/head:pull/13390` \
`$ git checkout pull/13390`

Update a local copy of the PR: \
`$ git checkout pull/13390` \
`$ git pull https://git.openjdk.org/jdk.git pull/13390/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13390`

View PR using the GUI difftool: \
`$ git pr show -t 13390`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13390.diff">https://git.openjdk.org/jdk/pull/13390.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13390#issuecomment-1500585413)

### CODE REVIEW

**Review by coleenp — APPROVED** [review]
Seems like a good thing to do.

File: src/hotspot/share/oops/instanceKlass.cpp [hunk 1]
```diff
@@ -2624,6 +2624,16 @@ void InstanceKlass::init_shared_package_entry() {
 #endif
 }
 
+void InstanceKlass::compute_has_loops_flag_for_methods() {
+  Array<Method*>* methods = this->methods();
+  for (int index = 0; index < methods->length(); ++index) {
+    Method *m = methods->at(index);
```
Comment (calvinccheung) [inline src/hotspot/share/oops/instanceKlass.cpp:2630]: Suggestion: Method* m = ...

**Review by calvinccheung — APPROVED** [review]
Looks good. Just one nit.

**Comment by iklam** [comment]
Thanks @coleenp and @calvinccheung for the review.
/integrate

**Pull request closed without merging**